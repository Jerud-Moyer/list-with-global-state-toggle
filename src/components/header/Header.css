.Header {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  align-items: center;

}
button {
  height: 50px;
  background-color:midnightblue;
  color: lightblue;
  border: 3px solid lightblue;
}
