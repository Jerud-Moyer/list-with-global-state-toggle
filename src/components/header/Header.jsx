import React from 'react';
import { useThemeToggle } from '../../hooks/themes';
import styles from '../header/Header.css';


export default function Header() {
  const toggle = useThemeToggle();

  return (
    <div className={styles.Header}>
      <h1>BENDERdex</h1>
      <button  onClick={toggle}>Day / Night</button>
    </div>
  );
}
