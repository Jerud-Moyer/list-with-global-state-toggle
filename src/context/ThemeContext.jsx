import React from 'react';
import { useState } from 'react';

export const ThemeContext = React.createContext();

// eslint-disable-next-line react/prop-types
export const ThemeProvider = ({ children }) => {
  const [themeSelection, setThemeSelection] = useState('day');

  const toggleTheme = () => {
    setThemeSelection(oldTheme => {
      if(oldTheme === 'day') return 'night';
      return 'day';
    });
  };
  
  
  return (
    <div>
      <ThemeContext.Provider value={{ themeSelection, toggleTheme }}>
        {children}
      </ThemeContext.Provider>
    </div>
  );

};
