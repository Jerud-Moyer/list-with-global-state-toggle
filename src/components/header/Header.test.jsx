import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Header from './Header';
import { ThemeProvider } from '../../context/ThemeContext';

describe('CharacterList component', () => {
  afterEach(() => cleanup());

  it('renders Character', () => {
    const { asFragment } = render(<ThemeProvider>
      <Header/>
    </ThemeProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
