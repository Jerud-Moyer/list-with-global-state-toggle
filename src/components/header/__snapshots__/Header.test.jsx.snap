// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterList component renders Character 1`] = `
<DocumentFragment>
  <div>
    <div
      class="Header"
    >
      <h1>
        BENDERdex
      </h1>
      <button>
        Day / Night
      </button>
    </div>
  </div>
</DocumentFragment>
`;
